Library/Homebrew/cli_parser.rb

```diff
@@ -0,0 +1,50 @@
+require "optparse"
+require "ostruct"
+
+module Homebrew
+  module CLI
+    class Parser
+      attr_accessor :parser
+
+      def initialize(&block)
+        @parser = OptionParser.new
+        @parsed_args = OpenStruct.new
+        instance_eval(&block)
+      end
+
+      def option(name, description: nil, type: nil, required: false)
```

Comment:
I think it would be a nicer API if `description` was just the second positional parameter, not a keyword argument. And, rather than a type parameter, it might be nicer to have different methods for different types of options? That would probably make the implementation simpler too.

So, the `audit` option definition would look something like this:

```ruby
switch       "--strict",             "strict"
switch       "--online",             "online"
switch       "--new-formula",        "new-formula"
switch       "--fix",                "fix"
switch       "--display-cop-names",  "display cop names"
switch       "--display-filename",   "display file names"

comma_array  "--only",               "only"
comma_array  "--except",             "except"
comma_array  "--only-cops",          "only cops"
comma_array  "--except-cops",        "except cops"
```

Reply:
Yeah, I was also thinking of having different methods! 👍